pyo3-macros-backend/src/method.rs

```diff
@@ -583,7 +583,8 @@ impl<'a> FnSpec<'a> {
                     ) -> _pyo3::PyResult<*mut _pyo3::ffi::PyObject> {
                         let function = #rust_name; // Shadow the function name to avoid #3017
                         #( #holders )*
-                        #call
+                        let result = #call;
+                        result
```

Comment:
These changes to return `result` on a separate line were caused by the fact that `#call` creates a borrow on `_args` and `_kwargs`, which now have `Drop` implementations due to being `Bound<PyTuple>` and `Option<Bound<PyDict>>` (if `*args`, or `**kwargs` are accepted).

So to satisfy the borrow checker we had to split the expression like this.

I think given our decision in macro code is to trend away from a single giant expression anyway, and it'll help me in #3847 where I need to do this too, this seemed a reasonable adjustment to me.